### PR TITLE
Reject block versions lower than 4

### LINKED ISSUE #1556: reject block versions < 4
The spec currently says that versions < 4 are rejected. We must either do that, or change the spec.

Note that code inherited from Bitcoin might do unintended things for lower block versions; by enforcing this rule, we would be reducing the number of code paths and improving auditability.

### PR DESCRIPTION
**Pull request by str4d**
Includes parts of #1554
Closes #1556

### CODE REVIEW

**Comment by str4d** [comment]
Merge #1600 before this PR.

File: src/chainparams.cpp [hunk 1]
```diff
@@ -280,11 +278,11 @@ class CRegTestParams : public CTestNetParams {
         nEquihashK = K;
         genesis.nTime = 1296688602;
```
Comment (daira) [inline src/chainparams.cpp]: This timestamp is 2011-02-02 23:16:42. Does not block (we're resetting again, I think).

Comment (daira) [inline src/chainparams.cpp]: Oh this is regtest, I meant to be commenting on the testnet timestamp.

File: src/chainparams.cpp [hunk 2]
```diff
@@ -190,11 +188,11 @@ class CTestNetParams : public CMainParams {
 
         //! Modify the testnet genesis block so the timestamp is valid for a later start.
         genesis.nTime = 1296688602;
```
Comment (daira) [inline src/chainparams.cpp]: This timestamp is 2011-02-02 23:16:42. Does not block (we're resetting again, I think).

**Comment by daira** [comment]
ut(ACK+cov). Need another ACK on my changes (https://github.com/str4d/zcash/compare/86d2c904684d3e4855f0d8998a950c9f758a461c...1556-reject-old-block-versions).

File: src/chainparams.cpp [hunk 3]
```diff
@@ -42,7 +42,6 @@ class CMainParams : public CChainParams {
         consensus.nMajorityEnforceBlockUpgrade = 750;
         consensus.nMajorityRejectBlockOutdated = 950;
         consensus.nMajorityWindow = 4000;
-        // TODO generate harder genesis block
```
Comment (str4d) [inline src/chainparams.cpp]: I was leaving these in since we may generate an even harder genesis block for 1.0.0, but does not block.

Comment (daira) [inline src/chainparams.cpp]: Right, but if for any reason we didn't do that, the current difficulty is sufficient so the comment is no longer needed, IMHO.

**Comment by daira** [comment]
@zkbot r+

**Comment by str4d** [comment]
Once #1600 is finished merging, this will succeed.

@zkbot retry

**Pull request merged**